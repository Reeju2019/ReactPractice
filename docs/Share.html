<!DOCTYPE html>
<html lang="en">

<head>
  
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title> Share</title>

  <script src="https://cdn.jsdelivr.net/gh/google/code-prettify@master/loader/run_prettify.js"></script>
  <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
  <script src="./build/entry.js"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
  <![endif]-->
  <link href="https://fonts.googleapis.com/css?family=Roboto:100,400,700|Inconsolata,700" rel="stylesheet">
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css" integrity="sha384-UHRtZLI+pbxtHCWp1t77Bi1L4ZtiqrqD80Kn4Z8NTSRyMA2Fd33n5dQ8lWUE00s/" crossorigin="anonymous">
  <link type="text/css" rel="stylesheet" href="https://jmblog.github.io/color-themes-for-google-code-prettify/themes/tomorrow-night.min.css">
  <link type="text/css" rel="stylesheet" href="styles/app.min.css">
  <link type="text/css" rel="stylesheet" href="styles/iframe.css">
  <link type="text/css" rel="stylesheet" href="">
  <script async defer src="https://buttons.github.io/buttons.js"></script>

  
</head>



<body class="layout small-header">
    <div id="stickyNavbarOverlay"></div>
    

<div class="top-nav">
    <div class="inner">
        <a id="hamburger" role="button" class="navbar-burger" aria-label="menu" aria-expanded="false">
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
            <span aria-hidden="true"></span>
        </a>
        <div class="logo">
            
             
                <a href="index.html">
                    <h1 class="navbar-item">My React components</h1>
                </a>
            
        </div>
        <div class="menu">
            
            <div class="navigation">
                <a
                    href="index.html"
                    class="link"
                >
                    API Documentation
                </a>
                
                
                
            </div>
        </div>
    </div>
</div>
    <div id="main">
        <div
            class="sidebar "
            id="sidebarNav"
        >
            
            <nav>
                
                    <h2><a href="index.html">Documentation</a></h2><div class="category"><h3>Classes</h3><ul><li><a href="AlertBox.html">AlertBox</a></li><li><a href="App.html">App</a></li><li><a href="BeforeAfterComponent.html">BeforeAfterComponent</a></li><li><a href="BeforeAfterSliderComponent.html">BeforeAfterSliderComponent</a></li><li><a href="CarousalItemComponent.html">CarousalItemComponent</a></li><li><a href="CarouselComponent.html">CarouselComponent</a></li><li><a href="CertificateComponent.html">CertificateComponent</a></li><li><a href="Certification.html">Certification</a></li><li><a href="ColumnChart.html">ColumnChart</a></li><li><a href="DoughnutChart.html">DoughnutChart</a></li><li><a href="EmailSelectorComponent.html">EmailSelectorComponent</a></li><li><a href="FailureModal.html">FailureModal</a></li><li><a href="Footer.html">Footer</a></li><li><a href="Header.html">Header</a></li><li><a href="ListGroupComponent.html">ListGroupComponent</a></li><li><a href="PdfGen.html">PdfGen</a></li><li><a href="ReportTable.html">ReportTable</a></li><li><a href="Share.html">Share</a></li><li><a href="SuccessToast.html">SuccessToast</a></li><li><a href="TemplateCard.html">TemplateCard</a></li><li><a href="UploadAsset.html">UploadAsset</a></li><li><a href="UploadBox.html">UploadBox</a></li><li><a href="UploadCard.html">UploadCard</a></li></ul><h3>Interfaces</h3><ul><li><a href="AlertBox.IProps.html">IProps</a></li><li><a href="AlertBox.IState.html">IState</a></li><li><a href="App.IProps.html">IProps</a></li><li><a href="App.IState.html">IState</a></li><li><a href="BeforeAfterSliderComponent.IProps.html">IProps</a></li><li><a href="BeforeAfterSliderComponent.IState.html">IState</a></li><li><a href="CarousalItemComponent.IProps.html">IProps</a></li><li><a href="CarousalItemComponent.IState.html">IState</a></li><li><a href="CarouselComponent.IProps.html">IProps</a></li><li><a href="CarouselComponent.IState.html">IState</a></li><li><a href="CertificateComponent.IProps.html">IProps</a></li><li><a href="CertificateComponent.IState.html">IState</a></li><li><a href="Certification.IProps.html">IProps</a></li><li><a href="Certification.IState.html">IState</a></li><li><a href="ColumnChart.IProps.html">IProps</a></li><li><a href="ColumnChart.IState.html">IState</a></li><li><a href="DoughnutChart.IProps.html">IProps</a></li><li><a href="DoughnutChart.IState.html">IState</a></li><li><a href="EmailSelectorComponent.IProps.html">IProps</a></li><li><a href="EmailSelectorComponent.IState.html">IState</a></li><li><a href="FailureModal.IProps.html">IProps</a></li><li><a href="FailureModal.IState.html">IState</a></li><li><a href="IProps.html">IProps</a></li><li><a href="ListGroupComponent.IProps.html">IProps</a></li><li><a href="ListGroupComponent.IState.html">IState</a></li><li><a href="PdfGen.IProps.html">IProps</a></li><li><a href="PdfGen.IState.html">IState</a></li><li><a href="ReportTable.IProps.html">IProps</a></li><li><a href="ReportTable.IState.html">IState</a></li><li><a href="Share.IProps.html">IProps</a></li><li><a href="Share.IState.html">IState</a></li><li><a href="SuccessToast.IProps.html">IProps</a></li><li><a href="SuccessToast.IState.html">IState</a></li><li><a href="TemplateCard.IProps.html">IProps</a></li><li><a href="TemplateCard.IState.html">IState</a></li><li><a href="UploadAsset.IProps.html">IProps</a></li><li><a href="UploadAsset.IState.html">IState</a></li><li><a href="UploadBox.IProps.html">IProps</a></li><li><a href="UploadBox.IState.html">IState</a></li><li><a href="UploadCard.IProps.html">IProps</a></li><li><a href="UploadCard.IState.html">IState</a></li></ul></div>
                
            </nav>
        </div>
        <div class="core" id="main-content-wrapper">
            <div class="content">
                <header class="page-title">
                    <p>Class</p>
                    <h1>Share</h1>
                </header>
                




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Share<span class="signature">(props)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        
            <div class='vertical-section'>
                <div class="members">
                    <div class="member">
                        <div class=name>
                            <span class="tag">Constructor</span>
                        </div>
                        


    
    <h4 class="name" id="Share">
        <a class="href-link" href="#Share">#</a>
        
        <span class="code-name">
            
                new Share<span class="signature">(props)</span><span class="type-signature"></span>
            
        </span>
    </h4>
    

    
    
    <div class="description">
        Share Component
    </div>
    









    <h5>Parameters:</h5>
    
<div class="table-container">
    <table class="params table">
        <thead>
        <tr>
            
            <th>Name</th>
            

            <th>Type</th>

            

            

            <th class="last">Description</th>
        </tr>
        </thead>

        <tbody>
        

            
<tr class="deep-level-0">
  
      <td class="name"><code>props</code></td>
  

  <td class="type">
  
      
<span class="param-type"><a href="IProps.html">IProps</a></span>


  
  </td>

  

  

  <td class="description last">Props</td>
</tr>


        
        </tbody>
    </table>
</div>





<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_Certification_CertificateComponent_Share_Share.tsx.html" class="button">View Source</a>
            <span>
                <a href="components_Certification_CertificateComponent_Share_Share.tsx.html">components/Certification/CertificateComponent/Share/Share.tsx</a>, <a href="components_Certification_CertificateComponent_Share_Share.tsx.html#line10">line 10</a>
            </span>
        </p>
    
</dl>


















    <div class='columns method-parameter'>
        <div class="column is-2"><label>Returns:</label></div>
        <div class="column is-10">
            
                    

<div class="columns">
    
    <div class='param-desc column is-7'>void</div>
    
    
</div>

                
        </div>
    </div>





                    </div>
                </div>
            </div>
        
    
    </div>
    
    

    

    

    
        <h3 class="subsection-title">Interfaces</h3>

        <dl>
            <dt><a href="Share.IProps.html">IProps</a></dt>
            <dd></dd>
        
            <dt><a href="Share.IState.html">IState</a></dt>
            <dd></dd>
        </dl>
    

    

    

    

    
        <div class='vertical-section'>
            <h1>Methods</h1>
            <div class="members">
            
                <div class="member">


    
    <h4 class="name" id="handleShareModalClose">
        <a class="href-link" href="#handleShareModalClose">#</a>
        
        <span class="code-name">
            
                handleShareModalClose<span class="signature">()</span><span class="type-signature"> &rarr; {void}</span>
            
        </span>
    </h4>
    

    
    
    <div class="description">
        Share Modal Close Handler
    </div>
    













    <h5 class="subsection-title">Properties:</h5>

    
<div class="table-container">
    <table class="props table">
        <thead>
        <tr>
            
            <th>Name</th>
            

            <th>Type</th>

            

            

            <th class="last">Description</th>
        </tr>
        </thead>

        <tbody>
        
            
<tr class="deep-level-0">
    
        <td class="name"><code>handleShareModalClose</code></td>
    

    <td class="type">
    
        
<span class="param-type">function</span>


    
    </td>

    

    

    <td class="description last">Handle share modal close</td>
</tr>

        
        </tbody>
    </table>
</div>



<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_Certification_CertificateComponent_Share_Share.tsx.html" class="button">View Source</a>
            <span>
                <a href="components_Certification_CertificateComponent_Share_Share.tsx.html">components/Certification/CertificateComponent/Share/Share.tsx</a>, <a href="components_Certification_CertificateComponent_Share_Share.tsx.html#line30">line 30</a>
            </span>
        </p>
    
</dl>


















    <div class='columns method-parameter'>
        <div class="column is-2"><label>Returns:</label></div>
        <div class="column is-10">
            
                    

<div class="columns">
    
    
    <div class='column is-5 has-text-left'>
        <label>Type: </label>
        
<span class="param-type">void</span>


    </div>
    
</div>

                
        </div>
    </div>




</div>
            
                <div class="member">


    
    <h4 class="name" id="handleShareModalShow">
        <a class="href-link" href="#handleShareModalShow">#</a>
        
        <span class="code-name">
            
                handleShareModalShow<span class="signature">(event)</span><span class="type-signature"> &rarr; {void}</span>
            
        </span>
    </h4>
    

    
    
    <div class="description">
        Share Modal Show Handler
    </div>
    









    <h5>Parameters:</h5>
    
<div class="table-container">
    <table class="params table">
        <thead>
        <tr>
            
            <th>Name</th>
            

            <th>Type</th>

            

            

            <th class="last">Description</th>
        </tr>
        </thead>

        <tbody>
        

            
<tr class="deep-level-0">
  
      <td class="name"><code>event</code></td>
  

  <td class="type">
  
      
<span class="param-type">object</span>


  
  </td>

  

  

  <td class="description last">Event</td>
</tr>


        
        </tbody>
    </table>
</div>





    <h5 class="subsection-title">Properties:</h5>

    
<div class="table-container">
    <table class="props table">
        <thead>
        <tr>
            
            <th>Name</th>
            

            <th>Type</th>

            

            

            <th class="last">Description</th>
        </tr>
        </thead>

        <tbody>
        
            
<tr class="deep-level-0">
    
        <td class="name"><code>handleShareModalShow</code></td>
    

    <td class="type">
    
        
<span class="param-type">function</span>


    
    </td>

    

    

    <td class="description last">Handle share modal show</td>
</tr>

        
        </tbody>
    </table>
</div>



<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    
    

    

    

    
        <p class="tag-source">
            <a href="components_Certification_CertificateComponent_Share_Share.tsx.html" class="button">View Source</a>
            <span>
                <a href="components_Certification_CertificateComponent_Share_Share.tsx.html">components/Certification/CertificateComponent/Share/Share.tsx</a>, <a href="components_Certification_CertificateComponent_Share_Share.tsx.html#line39">line 39</a>
            </span>
        </p>
    
</dl>


















    <div class='columns method-parameter'>
        <div class="column is-2"><label>Returns:</label></div>
        <div class="column is-10">
            
                    

<div class="columns">
    
    
    <div class='column is-5 has-text-left'>
        <label>Type: </label>
        
<span class="param-type">void</span>


    </div>
    
</div>

                
        </div>
    </div>




</div>
            
            </div>
        </div>
    

    

    
</article>

</section>




            </div>
            
            <footer class="footer">
                <div class="content has-text-centered">
                    <p>Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.6.7</a></p>
                    <p class="sidebar-created-by">
                        <a href="https://github.com/SoftwareBrothers/better-docs" target="_blank">BetterDocs theme</a> provided with <i class="fas fa-heart"></i> by 
                        <a href="http://softwarebrothers.co" target="_blank">SoftwareBrothers - JavaScript Development Agency</a>
                    </p>
                </div>
            </footer>
            
        </div>
        <div id="side-nav" class="side-nav">
        </div>
    </div>
<script src="scripts/app.min.js"></script>
<script>PR.prettyPrint();</script>
<script src="scripts/linenumber.js"> </script>

</body>
</html>