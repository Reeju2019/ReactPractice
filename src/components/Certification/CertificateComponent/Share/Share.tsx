import React from "react";

import { Button, Modal, Col } from "react-bootstrap";
import {
  FacebookShareButton,
  FacebookIcon,
  TwitterShareButton,
  TwitterIcon,
} from "react-share";

/**
 * @description Props of Share Component
 * @memberof Share
 * @property {string} shareUrl Sets the share url
 * @property {string} header Sets the header text
 * @property {string} closeButtonName Set the close button name
 * @property {string} shareButtonName Set the share button name
 * @property {Array} options Options
 */
/* eslint-disable */
interface IProps {
  shareUrl: string;
  header: string;
  closeButtonName: string;
  shareButtonName: string;
  options: string[];
}

/**
 * @description State of Share Component
 * @memberof Share
 * @property {boolean} shareModalShow Share modal show
 */
/* eslint-disable */
interface IState {
  shareModalShow: boolean;
}

/**
 * @component
 * This component was used in Certificate component
 * @description Share Component
 */
class Share extends React.Component<IProps, IState> {
  state: IState = {
    shareModalShow: false,
  };

  /**
   * @constructor
   * @description Constructor
   * @param {IProps} props Props
   * @returns void
   */
  constructor(props: IProps) {
    super(props);
    this.handleShareModalClose = this.handleShareModalClose.bind(this);
    this.handleShareModalShow = this.handleShareModalShow.bind(this);
  }

  /**
   * Share Modal Close Handler
   * @property {Function} handleShareModalClose Handle share modal close
   * @returns {void}
   */
  handleShareModalClose() {
    this.setState({ shareModalShow: false });
  }

  /**
   * Share Modal Show Handler
   * @property {Function} handleShareModalShow Handle share modal show
   * @param {object} event Event
   * @returns {void}
   */
  handleShareModalShow(event) {
    const classLists = event.target.classList;
    for (const key in classLists) {
      if (classLists[key] === "share-btn")
        this.setState({ shareModalShow: true });
    }
  }

  render() {
    return (
      <>
        <Col
          xs={4}
          md={1}
          className="share-btn"
          onClick={this.handleShareModalShow}
        >
          {this.props.shareButtonName}
          <Modal
            className="d-flex justify-content-center p-5"
            show={this.state.shareModalShow}
            onHide={this.handleShareModalClose}
          >
            <Modal.Header>
              <Modal.Title>{this.props.header}</Modal.Title>
            </Modal.Header>
            <Modal.Body className="d-flex justify-content-center">
              {this.props.options.includes("facebook") ? (
                <FacebookShareButton url={this.props.shareUrl}>
                  <FacebookIcon />
                </FacebookShareButton>
              ) : null}
              {this.props.options.includes("twitter") ? (
                <TwitterShareButton url={this.props.shareUrl}>
                  <TwitterIcon />
                </TwitterShareButton>
              ) : null}
            </Modal.Body>
            <Modal.Footer>
              <Button variant="secondary" onClick={this.handleShareModalClose}>
                {this.props.closeButtonName}
              </Button>
            </Modal.Footer>
          </Modal>
        </Col>
      </>
    );
  }
}

export { Share };
export default Share;
