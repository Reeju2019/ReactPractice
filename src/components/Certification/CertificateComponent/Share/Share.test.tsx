import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import Share from "./Share"

/**
 * Unit testing for Share Component
 */
describe("Share Component Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <Share
        shareUrl={"xyz.com"}
        header={"header"}
        closeButtonName={"close"}
        shareButtonName={"share"}
        options={["facebook", "twitter"]}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <Share
        shareUrl={"xyz.com"}
        header={"header"}
        closeButtonName={"close"}
        shareButtonName={"share"}
        options={["facebook", "twitter"]}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
