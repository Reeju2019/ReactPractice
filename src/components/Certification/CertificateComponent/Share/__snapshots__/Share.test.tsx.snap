// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Component Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Col
      className="share-btn"
      md={1}
      onClick={[Function]}
      xs={4}
    >
      share
      <Modal
        animation={true}
        autoFocus={true}
        backdrop={true}
        className="d-flex justify-content-center p-5"
        dialogAs={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        enforceFocus={true}
        keyboard={true}
        onHide={[Function]}
        restoreFocus={true}
        show={false}
      >
        <ModalHeader
          closeButton={false}
          closeLabel="Close"
        >
          <ModalTitle>
            header
          </ModalTitle>
        </ModalHeader>
        <ModalBody
          className="d-flex justify-content-center"
        >
          <ForwardRef(ShareButton-facebook)
            url="xyz.com"
          >
            <Icon
              bgStyle={Object {}}
              borderRadius={0}
              iconFillColor="white"
              size={64}
            />
          </ForwardRef(ShareButton-facebook)>
          <ForwardRef(ShareButton-twitter)
            url="xyz.com"
          >
            <Icon
              bgStyle={Object {}}
              borderRadius={0}
              iconFillColor="white"
              size={64}
            />
          </ForwardRef(ShareButton-twitter)>
        </ModalBody>
        <ModalFooter>
          <Button
            active={false}
            disabled={false}
            onClick={[Function]}
            variant="secondary"
          >
            close
          </Button>
        </ModalFooter>
      </Modal>
    </Col>
  </React.Fragment>,
]
`;
